File: 9-PalindromeNumber/9-PalindromeNumber.py
# Last updated: 4/8/2026, 5:14:19 PM
class Solution:
    def isPalindrome(self, x: int) -> bool:
        if x < 0:
            return False
        if x < 10:
            return True
        if x % 10 == 0:
            return False
            
        reverted_number = 0
        while x > reverted_number:
            reverted_number = reverted_number * 10 + x % 10
            x //= 10
        
        return x == reverted_number or x == reverted_number // 10

File: 9-PalindromeNumber/9-PalindromeNumber.cpp
// Last updated: 4/8/2026, 5:14:21 PM
class Solution {
public:
    bool isPalindrome(int x) {
        std::string y = std::to_string(x);
        for (int i = 0; i < y.length() / 2; i++) {
            if (y[i] != y[y.length() - 1 - i]) {
                return false;
            }
        }
        return true;
    }
}; 